src/components/developer-options.tsx

```diff
@@ -1,84 +1,108 @@
-import React, { useEffect, useState } from "react";
-import { OpenAI } from "openai";
-import { observer } from "mobx-react-lite";
+import React from "react";
 import { AssistantModelType } from "../models/assistant-model";
 import { useAppConfigContext } from "../hooks/use-app-config-context";
-import { useOpenAIContext } from "../hooks/use-openai-context";
-
-import "./developer-options.scss";
+import { getUrlParam } from "../utils/utils";
+import { DAVAI_SPEAKER, GREETING } from "../constants";
+import { IUserOptions } from "../types";
+import { observer } from "mobx-react-lite";
 
 interface IProps {
+  createToggleOption: (option: keyof IUserOptions, optionLabel: string) => React.JSX.Element;
   assistantStore: AssistantModelType;
-  onCreateThread: () => void;
-  onDeleteThread: () => void;
-  onSelectAssistant: (id: string) => void;
 }
 
-export const DeveloperOptionsComponent = observer(function DeveloperOptions({assistantStore, onCreateThread, onDeleteThread, onSelectAssistant}: IProps) {
+export const DeveloperOptionsComponent = observer(({assistantStore, createToggleOption}: IProps) => {
   const appConfig = useAppConfigContext();
-  const apiConnection = useOpenAIContext();
   const selectedAssistant = assistantStore.assistantId ? assistantStore.assistantId : "mock";
-  const [assistantOptions, setAssistantOptions] = useState<Map<string, string>>();
+  const isDevMode = getUrlParam("mode") === "development" || appConfig.mode === "development";
 
-  useEffect(() => {
-    const fetchAssistants = async () => {
-      try {
-        const res = await apiConnection.beta.assistants.list();
-        const assistants = new Map();
-        res.data.map((assistant: OpenAI.Beta.Assistant) => {
-          const assistantName = assistant.name || assistant.id;
-          assistants.set(assistant.id, assistantName);
-        });
-        setAssistantOptions(assistants);
-      } catch (err) {
-        console.error(err);
-      }
-    };
+  const handleCreateThread = async () => {
```

Comment:
Since we were already passing down the `assistantStore` to this component, I figured we could move these functions that were previously defined in `App` to avoid more prop-drilling.